scanners/nodejs/utiltity.go

```diff
@@ -0,0 +1,352 @@
+package nodejs
+
+import (
+	"fmt"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/bitrise-io/bitrise-init/models"
+	"github.com/bitrise-io/bitrise-init/steps"
+	"github.com/bitrise-io/bitrise-init/utility"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type nodeVersionResult struct {
+	version string
+	file    string
+}
+
+type checkScriptResult struct {
+	scripts                    []string
+	hasBuild, hasLint, hasTest bool
+}
+
+// Detection
+func checkPackageJSON(searchDir string) (string, error) {
+	log.TPrintf("Checking package.json")
+	packageJsonPath := filepath.Join(searchDir, packageJson)
+	if exists := utility.FileExists(packageJsonPath); exists {
+		log.TPrintf("- %s - found", packageJsonPath)
+		return filepath.Rel(searchDir, packageJsonPath)
+	}
+
+	log.TPrintf("- %s - not found", packageJsonPath)
+	return "", fmt.Errorf("package.json not found")
+}
+
+var nodeVersionFiles = []string{".nvmrc", ".node-version", ".tool-versions"}
+
+func checkNodeVersion(searchDir string) nodeVersionResult {
+	log.TPrintf("Checking Node version")
+
+	result := nodeVersionResult{
+		version: "",
+		file:    "",
+	}
+
+	for _, fileName := range nodeVersionFiles {
+		versionFilePath := filepath.Join(searchDir, fileName)
+		hasVersionFile := utility.FileExists(versionFilePath)
+
+		if !hasVersionFile {
+			log.TPrintf("- %s - not found", fileName)
+			continue
+		}
+
+		log.TPrintf("- %s - found", fileName)
+		fileContent, err := fileutil.ReadStringFromFile(versionFilePath)
+		if err != nil {
+			log.TWarnf("Failed to read node version from %s", fileName)
+			continue
+		}
+
+		nodeVersion := strings.TrimSpace(fileContent)
+		if fileName == ".tool-versions" {
+			nodeVersion = getNodeVersionFromToolVersions(fileContent)
+		}
+
+		log.TPrintf("Node version: %s", nodeVersion)
+
+		return nodeVersionResult{
+			version: nodeVersion,
+			file:    fileName,
+		}
+	}
+
+	return result
+}
+
+func getNodeVersionFromToolVersions(fileContent string) string {
+	lines := strings.Split(fileContent, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) >= 1 && fields[0] == "nodejs" {
+			return fields[1]
+		}
```

Comment:
If the `.tool-versions` file contains only `nodejs` then this will crash. I guess that is not a correct file content format but still we should guard against it.